Add grid to the doc start page (#353)
ansys/pyhps: doc/source/index.rst

* Monitor and manage jobs.
* Run your own design exploration algorithms.
* Retrieve simulation results.

.. grid:: 1 1 2 2
        :gutter: 2

        .. grid-item-card:: :octicon:`rocket` Getting started
            :link: getting_started/index
            :link-type: doc

>             Contains installation instructions to get you
>             started with PyHPS.

Reviewer: ```suggestion
            Learn how to install and start using PyHPS.
```

        .. grid-item-card:: :octicon:`tools` User guide
            :link: user_guide/index
            :link-type: doc

>             Walks you through the basics of how to interact with HPS.

Reviewer: ```suggestion
            Understand the basics of how to interact with HPS.
```

        .. grid-item-card:: :octicon:`play` Examples
            :link: examples/index
            :link-type: doc

>             A collection of examples demonstrating how to submit jobs using PyHPS.

Reviewer: ```suggestion
            Explore examples demonstrating how to use PyHPS to submit jobs.
```

        .. grid-item-card:: :octicon:`file-code` API reference
            :link: api/index
            :link-type: doc

>             Describes the public Python classes, methods, and functions.

Author: I've take the freedom to rephrase it as 

> Understand the PyHPS API, its capabilities, and how to interact with it programmatically.

(it's unusual to talk about Python API endpoints, it's more a terminology for REST APIs)
